src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -55,15 +56,29 @@ class Crawler implements \Countable, \IteratorAggregate
     private $isHtml = true;
 
     /**
-     * @param mixed  $node     A Node to use as the base for the crawling
-     * @param string $uri      The current URI
-     * @param string $baseHref The base href value
+     * @var HTML5|null
      */
-    public function __construct($node = null, string $uri = null, string $baseHref = null)
+    private $html5Parser;
+
+    /**
+     * @param mixed     $node           A Node to use as the base for the crawling
+     * @param string    $uri            The current URI
+     * @param string    $baseHref       The base href value
+     * @param bool|null $useHtml5Parser Whether the Crawler should use the HTML5 parser or the native DOM parser
+     */
+    public function __construct($node = null, string $uri = null, string $baseHref = null, bool $useHtml5Parser = null)
     {
         $this->uri = $uri;
         $this->baseHref = $baseHref ?: $uri;
 
+        if ($useHtml5Parser && !class_exists(HTML5::class)) {
+            throw new \LogicException('Using the DomCrawler HTML5 parser requires the html5-php library. Try running "composer require masterminds/html5".');
+        }
+
+        if ($useHtml5Parser ?? class_exists(HTML5::class)) {
+            $this->html5Parser = new HTML5(['disable_html_ns' => true]);
```

Comment:
When creating a child crawler, you should not rely on guessing but pass the existing value used for the parsing (or even better, assign the actual parser instead of instantiating a new one).

Reply:
You mean in the `createSubCrawler` method?

Reply:
yes